### ISSUE TITLE
install fails when rest config is set

### ISSUE THREAD

**Issue by kwmonroe** [juju-solutions/layer-apache-zookeeper#7]
This works:

```
juju deploy apache-zookeeper zk
# wait till it settles to "Ready"
juju set-config zk rest=True
```

This fails:

```
juju deploy apache-zookeeper zk2
# don't wait, set config immediately afer deployment so it is set before install() fires
juju set-config zk2 rest=True
```

Install will fail like this:

http://paste.ubuntu.com/16665517/

To make matters worse, the hook is automatically retried, which leads to running install() a second time.  This is very problematic because we do not handle re-running the /usr/lib/zookeeper/conf handler correctly and fail like this (picking up with a few lines overlap from the above failure):

http://paste.ubuntu.com/16665543/

So to summarize, we need to fix and/or handle whatever is causing multiple REST servers to start when rest=True prior to install.  It would be nice to fix the other thing too, so install() was idempotent.

**Comment by kwmonroe** [comment]
One more thing, there was an earlier failure that would happen when trying to `utils.run_as(pkill -f RestMain)`, but that is fixed here:

https://github.com/juju-solutions/layer-apache-zookeeper/commit/64c27a1c463400a26a0b1fd4597906ed171f6e01

The fix isn't quite enough though, since it now fails with too many RestMains running.
